src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -167,6 +168,14 @@ public void AddScheme(AuthenticationScheme scheme)
                 {
                     _requestHandlers.Add(scheme);
                 }
+                if (typeof(IAuthenticationSignInHandler).IsAssignableFrom(scheme.HandlerType))
+                {
+                    _signInHandlers.Add(scheme);
+                }
+                if (typeof(IAuthenticationSignOutHandler).IsAssignableFrom(scheme.HandlerType))
+                {
+                    _signOutHandlers.Add(scheme);
+                }
                 _map[scheme.Name] = scheme;
```

Comment:
There's no check that the HandlerType actually implements IAuthenticationHandler